### PR TITLE
TASK: Decouple Package Key and Node type namespace for localization

### PR DESCRIPTION
**Pull request by dfeyer**
This change make possible to create node type in package A with a node
type namespace B with the localization support. Without this change
only node type with namespace A (so coupled with the package key) are
allowed.

### CODE REVIEW

File: Neos.Neos/Classes/Service/XliffService.php [hunk 1]
```diff
@@ -175,9 +177,6 @@ function (\XMLReader $file, $offset, $version) use ($packageKey, &$sources, $sou
                         $targetPackageKey = $file->getAttribute('product-name') ?: $packageKey;
                         $source = $file->getAttribute('original') ?: $source;
                     }
-                    if ($packageKey !== $targetPackageKey) {
```
Comment (dfeyer) [inline Neos.Neos/Classes/Service/XliffService.php]: This basically invalidate the comment "in xliff v1.2 the packageKey or source can be overwritten via attributes", if the packageKey is overwritten, the localization are just skipped completly.

**Comment by dfeyer** [comment]
Maybe this can be a bugfix, and need to be done in branch 4.0, but I'm intersted to have more inputs from @DrillSergeant about this, not sure I understand the code correctly.

File: Neos.Neos/Classes/Service/XliffService.php (hunk 1 above)
Comment (DrillSergeant) [reply Neos.Neos/Classes/Service/XliffService.php]: I think only translationSources for the given packageKey should be collected in the method collectPackageSources. If sources are found for an other packageKey they are skipped which makes sense to me.
If you have xlf-Files in your package with "product-name" pointing to another package, they dont need to be loaded here, but they will be loaded when the destination package is getting loaded.
TLDR, I think the line is correct. Do you agree?

Comment (dfeyer) [reply Neos.Neos/Classes/Service/XliffService.php]: > If you have xlf-Files in your package with "product-name" pointing to another package, they dont need to be loaded here, but they will be loaded when the destination package is getting loaded.

That's my concern ... so the package key is coupled with the node type namespace. Maybe it's for good reasons, but I'm not convinced ;) 

It's the only place where both concept are coupled. You can define a node type Foo.Vendor:MyDocument in package Vendor.Website:Content and the content repository or fusion works fine, there is just the translation that does not works.

As I'm not fully confident with the concept of overriding translation between package ... did this PR break the current implementation ?

Maybe we need @kitsunet in the discussion ?

Comment (dfeyer) [reply Neos.Neos/Classes/Service/XliffService.php]: An other question, did this (custom node type namespace in arbitrary package, with translation) works before we implement the translation overriding ? ... Need to check this, we implement this feature in 4.0 right ?
